mlir/lib/Dialect/MIOpen/Generator/Conv2dGenerator.cpp

```diff
@@ -590,3 +590,71 @@ LogicalResult Conv2dGenerator::genConvModule(ModuleOp &module,
 
   return success();
 }
+
+int64_t Conv2dGenerator::getBwdDataNumberOfGemm() const {
+  auto gcdStrideDilationH =
+      math_util::gcd(config.strideHeight, config.dilationHeight);
+  auto gcdStrideDilationW =
+      math_util::gcd(config.strideWidth, config.dilationWidth);
+
+  auto yTilda = config.strideHeight / gcdStrideDilationH;
+  auto xTilda = config.strideWidth / gcdStrideDilationW;
+  return yTilda * xTilda;
+}
+
+std::tuple<int64_t, int64_t, int64_t>
+Conv2dGenerator::getBwdDataGemmSizes(int64_t gemmId) const {
```

Comment:
I'm confused:

- Does this apply only to backward data? 
- What's special about bwd?
- Why do we want to do gemm size calculation for bwd? Since I think this will be done in deriving the tuning parameter anyways?